src/Castle.Windsor/Compatibility/ApplicationDomains.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace Castle.Core.Internal
+{
+    internal static class ApplicationDomains
```

Comment:
I'm thinking this would be better named `AppContext` and conditionally compiled only for .NET 4.5, so that when we rev to .NET 4.6 in the future this code doesn't have to be yanked out.